src/lib.rs

```diff
@@ -3133,6 +3135,65 @@ pub trait Itertools: Iterator {
         self.k_largest_by(k, k_smallest::key_to_cmp(key))
     }
 
+    /// Consumes the iterator and return an iterator of the last `n` elements.
+    ///
+    /// It allocates up to `n` elements.
+    /// The iterator, if directly collected to a `VecDeque`, is converted
+    /// without any extra copying or allocation cost.
+    /// If directly collected to a `Vec`, it may need some data movement
+    /// but no re-allocation.
```

Comment:
You might want to leave some leeway for the implementation here. For example allocating a `VecDeque` with twice the necessary capacity will allow you to do the rotating in batches or use `next_chunk` to overfill it and then trim it back.

Reply:
Allocating twice the necessary capacity definitely improved my benchmark for a `VecDeque`, but it still remains slower the current `Vec` version.

Reply:
I was more talking about the general case, not documentating too much about today's implementation to leave room for future improvements.

Reply:
Fair enough. Just about "_allocates up to `n`_" I suppose.